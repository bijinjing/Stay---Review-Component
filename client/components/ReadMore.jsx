import React from 'react';
import styled from 'styled-components'

const Read_More = styled.a`
  font:inherit
  color:#008489
  cursor: pointer
  background: transparent
  -webkit-font-smoothing: antialiase
`;



class ReadMore extends React.Component{
  constructor(props) {
    super(props)

    this.state={
      expand:false
    }

    this.expandHandler = this.expandHandler.bind(this)
  }

  expandHandler(){
    let oldState = this.state.expand;
    let expand = oldState? false: true
    this.setState({
      expand
    })
  }

  render(){
    return(
       this.state.expand? <div id="read_more_false">{this.props.comments}</div>:
        <div id="read_more_true">
          <a>{this.props.comments.split(" ").slice(0,20).join(" ") + "..."}</a><Read_More id="read_more_button" onClick={this.expandHandler} style={{}}>Read more</Read_More>
        </div>
    )
  }
}

export default ReadMore