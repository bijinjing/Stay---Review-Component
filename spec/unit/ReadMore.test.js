import React from 'react';
import ReadMore from '../../client/components/ReadMore.jsx';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Enzyme from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import {mount} from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });

describe('Search', ()=>{

  it('renders without crashing when expand is true', () => {
    const props = {
      expand:true,
      comments:"erw qertr fshfs era fsgfdhf gfg eyrrw qetewr dg jkl eqt jkl yu jk fghjk kl; hjk hnjmk, ghj hjk hjk hjk wer sdfg ikjhb gfc hgvc iujh red ygfv okj plkm tgv ed yhjkl sdfghj asdfghj werty asdfg fghj fghjk uio tyuik "
    }

    const wrapper = shallow(<ReadMore {...props}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders without crashing when expand is false', () => {
    const props = {
      expand:false,
      comments:"erw qertr fshfs era fsgfdhf gfg eyrrw"
    }

    const wrapper = shallow(<ReadMore {...props}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })


  it('should run expandHandler with clicking on Read more', () => {
    let result = null;
    const props = {
      expand:true,
      comments:"erw qertr fshfs era fsgfdhf gfg eyrrw qetewr dg jkl eqt jkl yu jk fghjk kl ",
      expandHandler:()=>{ this.setState({expand:false})}
    }
    const component = mount(<ReadMore {...props}/>);
    component
      .find('#read_more_button')
      .simulate('click');

    expect(component.find('#read_more_false').length).toEqual(1);
    component.unmount();
  });
})