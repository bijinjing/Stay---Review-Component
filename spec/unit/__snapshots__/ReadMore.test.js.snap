// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders without crashing when expand is false 1`] = `
<div
  id="read_more_true"
>
  <a>
    erw qertr fshfs era fsgfdhf gfg eyrrw...
  </a>
  <ReadMore__Read_More
    id="read_more_button"
    onClick={[Function]}
    style={Object {}}
  >
    Read more
  </ReadMore__Read_More>
</div>
`;

exports[`Search renders without crashing when expand is true 1`] = `
<div
  id="read_more_true"
>
  <a>
    erw qertr fshfs era fsgfdhf gfg eyrrw qetewr dg jkl eqt jkl yu jk fghjk kl; hjk hnjmk, ghj hjk...
  </a>
  <ReadMore__Read_More
    id="read_more_button"
    onClick={[Function]}
    style={Object {}}
  >
    Read more
  </ReadMore__Read_More>
</div>
`;
